### PR TITLE
Re-enable new ambiguity warning under 2.13

### PR DESCRIPTION
**Pull request by lrytz**
Include a quick fix.

follow-up for https://github.com/scala/scala/pull/10339

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/Typers.scala [hunk 1]
```diff
@@ -5568,8 +5568,11 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
               case sym      => typed1(tree setSymbol sym, mode, pt)
             }
           case LookupSucceeded(qual, sym)   =>
-            sym.getAndRemoveAttachment[LookupAmbiguityWarning].foreach(w =>
-              runReporting.warning(tree.pos, w.msg, WarningCategory.Other, context.owner))
+            sym.getAndRemoveAttachment[LookupAmbiguityWarning].foreach(w => {
+              val cat = if (currentRun.isScala3) WarningCategory.Scala3Migration else WarningCategory.Other
+              val fix = List(CodeAction("ambiguous reference", Some(w.msg), List(TextEdit(tree.pos.focusStart, w.fix))))
+              runReporting.warning(tree.pos, w.msg, cat, context.owner, fix)
+            })
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/Typers.scala:5575]: Maybe the way to motivate `something.foreach { w => block }` syntax is to say, the lambda is a block, much like "anonymous pattern-matching function" is a block (of cases). For this syntax, it introduces a parameter. How slick! Or maybe, "it's like `{ case w => block }` but you can omit the `case`. Hey!

**Review by SethTisue — APPROVED** [review]

**Comment by SethTisue** [comment]
@lrytz we don't need to release-note this... or do we?

**Comment by som-snytt** [comment]
I was hoping to read an explanation. Oh, is this to warn without `-Xsource:3`? We should release-note that they should be using `-Xsource:3`. Also, is there any truth to the rumor that `-X` will be renamed `-twitter`?

**Comment by lrytz** [comment]
The new warning has a very detailed message, so it probably doesn't need to be mentioned in the release notes.

**Pull request merged**